keel-web/src/main/kotlin/com/netflix/spinnaker/keel/services/ApplicationService.kt

```diff
@@ -48,21 +47,17 @@ class ApplicationService(
     resources = resources.map { summary ->
       summary.resource.let { resource ->
         if (resource.spec is ComputeResourceSpec) {
-          resolvers.supportingComputeResources(resource as Resource<ComputeResourceSpec>)
-            .fold(resource) { computeResource, resolver ->
-              log.debug("Applying ${resolver.javaClass.simpleName} to ${computeResource.id}")
-              resolver(computeResource)
-            }.let { computeResource ->
-              if (computeResource.spec.deliveryArtifact != null && computeResource.spec.artifactVersion != null) {
-                summary.copy(artifact = ResourceArtifactSummary(
-                  name = computeResource.spec.deliveryArtifact!!.name,
-                  type = computeResource.spec.deliveryArtifact!!.type,
-                  desiredVersion = computeResource.spec.artifactVersion!!
-                ))
-              } else {
-                summary
-              }
-            }
+          (resource.spec as ComputeResourceSpec).let {
+            val lastRelevantEvent =
+              repository.resourceLastEventByType(resource.id, ResourceValid::class.java)
```

Comment:
This gives preference to a `ResourceValid` event to get the artifact information. If the resource is not yet valid, we use the information from the `ResourceCreated` event.